main.c

```diff
@@ -38,6 +38,47 @@ void doArrDef(char type, int type_sz) {
   next();
 }
 
+void doDefString() {
+  next();
+  if(token != NAME) {
+    expected("variable name");
+  }
+  char name[100];
```

Comment:
yeah, you're right we should fix this, but I think it should be done in another issue.

The thing is this "name" buffer is 100 bytes long because the global "value" buffer is 100 bytes long as well, we should resize the value buffer to 256 and any other buffers that copy it's contents.